compose/service.py

```diff
@@ -246,6 +246,12 @@ def recreate_container(self, container, **override_options):
         self.client.rename(
             container.id,
             '%s_%s' % (container.short_id, container.name))
+
+        override_options = dict(
+            override_options,
+            environment=merge_environment(
+                override_options.get('environment'),
+                {'affinity:container': container.id}))
```

Comment:
Hmm, it needs to come out as `affinity:container==<id>`, i.e. a double-equals. Maybe we should add a test for it?

(I'm going to investigate running Compose's test suite against Swarm, btw)